value/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019-2022 Parity Technologies (UK) Ltd.
+// This file is part of subxt.
+//
+// subxt is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// subxt is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with subxt.  If not, see <http://www.gnu.org/licenses/>.
+
+//! This crate exposes the [`Value`] type and related subtypes, which are used as the runtime
+//! representations of SCALE encoded data (much like `serde_json::Value` is a runtime representation
+//! of JSON data).
+
+mod scale_impls;
+#[cfg(feature = "serde")]
+mod serde_impls;
+mod value_type;
+
+pub use value_type::{
+    BitSequence,
+    Composite,
+    Primitive,
+    Value,
+    ValueDef,
+    Variant,
+};
+
+/// Serializing and deserializing a [`crate::Value`] into/from other types via serde.
+#[cfg(feature = "serde")]
+pub mod serde {
+    pub use crate::serde_impls::DeserializeError;
+
+    /// Attempt to deserialize a [`crate::Value`] into another type.
+    pub fn from_value<'de, Ctx, T: serde::Deserialize<'de>>(
+        value: crate::Value<Ctx>,
+    ) -> Result<T, DeserializeError> {
+        T::deserialize(value)
+    }
+
+    // TODO: Eventually let's implement Serializer on Value so that we can convert from some type into a Value:
+    //
+    // /// Attempt to serialize some type into a [`crate::Value`].
+    // pub fn to_value<'de, Ctx, T: serde::Serialize>(ty: T) -> Result<crate::Value<()>, SerializeError> {
+    //     ty.serialize(serializer)
+    // }
```

Comment:
I think we can merge this PR without adding this Serialize impl, but it would be a nice to have to round out the interface (it's the only serde trait not implemented, and being able to cast from structs to intermediate Value types may be useful (for example in allowing more flexible encodings if there is a specific known type you'd like to encode to).